lib/src/request/RequestPlugin.ts

```diff
@@ -0,0 +1,10 @@
+import { CommonMediaRequest } from './CommonMediaRequest.js';
+
+/**
+ * Request plugin API.
+ * @param request - The request to be executed.
+ * @returns A promise that is resolved when the plugin has completed the manipulation of the request.
+ * 
+ * @group Request
+ */
+export type RequestPlugin = (request: CommonMediaRequest) => Promise<any>
```

Comment:
Design question: Should this function return a new/modified request, or should it modify the incoming request and return `void`?

Reply:
I would say it is more suitable and flexible to return the new or modified request.

`export type RequestPlugin = (request: CommonMediaRequest) => Promise<CommonMediaRequest>`

Reply:
Agreed.